stdlib/iarray.mli

```diff
@@ -0,0 +1,294 @@
+(** [map2 f a b] applies function [f] to all the elements of [a]
+   and [b], and builds an immutable array with the results returned by [f]:
+   [[| f (get a 0) (get b 0);
+       ...;
+       f (get a (length a - 1)) (get b (length b - 1))|]].
+   @raise Invalid_argument if the immutable arrays are not the same size. *)
+
+
+(** {1 Array scanning} *)
+
+val for_all : ('a -> bool) -> 'a iarray -> bool
+(** [for_all f [|a1; ...; an|]] checks if all elements
+   of the immutable array satisfy the predicate [f]. That is, it returns
+   [(f a1) && (f a2) && ... && (f an)]. *)
+
+val exists : ('a -> bool) -> 'a iarray -> bool
+(** [exists f [|a1; ...; an|]] checks if at least one element of
+    the immutable array satisfies the predicate [f]. That is, it returns
+    [(f a1) || (f a2) || ... || (f an)]. *)
+
+val for_all2 : ('a -> 'b -> bool) -> 'a iarray -> 'b iarray -> bool
+(** Same as {!for_all}, but for a two-argument predicate.
+   @raise Invalid_argument if the two immutable arrays have different
+   lengths. *)
+
+val exists2 : ('a -> 'b -> bool) -> 'a iarray -> 'b iarray -> bool
+(** Same as {!exists}, but for a two-argument predicate.
+   @raise Invalid_argument if the two immutable arrays have different
+   lengths. *)
+
+val mem : 'a -> 'a iarray -> bool
+(** [mem a set] is true if and only if [a] is structurally equal
+    to an element of [l] (i.e. there is an [x] in [l] such that
+    [compare a x = 0]). *)
+
+val memq : 'a -> 'a iarray -> bool
+(** Same as {!mem}, but uses physical equality instead of structural equality
+    to compare array elements. *)
+
+val find_opt : ('a -> bool) -> 'a iarray -> 'a option
+(** [find_opt f a] returns the first element of the immutable array [a] that
+    satisfies the predicate [f], or [None] if there is no value that satisfies
+    [f] in the array [a]. *)
+
+val find_index : ('a -> bool) -> 'a iarray -> int option
+(** [find_index f a] returns [Some i], where [i] is the index of the first
+    element of the array [a] that satisfies [f x], if there is such an
+    element.
+
+    It returns [None] if there is no such element. *)
+
+val find_map : ('a -> 'b option) -> 'a iarray -> 'b option
+(** [find_map f a] applies [f] to the elements of [a] in order, and returns the
+    first result of the form [Some v], or [None] if none exist. *)
+
+val find_mapi : (int -> 'a -> 'b option) -> 'a iarray -> 'b option
+(** Same as [find_map], but the predicate is applied to the index of
+   the element as first argument (counting from 0), and the element
+   itself as second argument. *)
+
+(** {1 Arrays of pairs} *)
+
+val split : ('a * 'b) iarray -> 'a iarray * 'b iarray
+(** [split [:(a1,b1); ...; (an,bn):]] is
+    [([:a1; ...; an:], [:b1; ...; bn:])]. *)
+
+val combine : 'a iarray -> 'b iarray -> ('a * 'b) iarray
+(** [combine [:a1; ...; an:] [:b1; ...; bn:]] is [[:(a1,b1); ...; (an,bn):]].
+    Raise [Invalid_argument] if the two immutable arrays have different
+    lengths. *)
```

Comment:
Unless I'm mistaken the syntax was not included in the PR. The doc string of these two functions still mention it.